libs/auth/src/artifact_authenticator.ts

```diff
@@ -0,0 +1,201 @@
+   */
+  async writeSignatureFile(artifact: Artifact): Promise<void> {
+    const message = await this.constructMessage(artifact);
+    const messageSignature = await this.signMessage(message);
+    const signingMachineCert = await fs.readFile(this.signingMachineCertPath);
+    const signatureFile = this.serializeSignatureFileContents({
+      signature: messageSignature,
+      signingMachineCert,
+    });
+    await fs.writeFile(
+      this.constructSignatureFilePath(artifact),
+      signatureFile
+    );
+  }
+
+  /**
+   * Verifies the authenticity of the provided artifact using its signature file, which is expected
+   * to be found at the same file path as the artifact, but with a .vxsig extension, e.g.
+   * /path/to/artifact.txt.vxsig.
+   */
+  async authenticateArtifactUsingSignatureFile(
+    artifact: Artifact
+  ): Promise<Result<void, Error>> {
+    try {
+      await this.authenticateArtifactUsingSignatureFileHelper(artifact);
+    } catch {
+      // TODO: Log raw error
+      return err(
+        new Error(`Error authenticating ${artifact.path} using signature file`)
+      );
+    }
+    return ok();
+  }
+
+  private async authenticateArtifactUsingSignatureFileHelper(
+    artifact: Artifact
+  ): Promise<void> {
+    const message = await this.constructMessage(artifact);
+
+    const signatureFile = await fs.readFile(
+      this.constructSignatureFilePath(artifact)
+    );
+    const { signature: messageSignature, signingMachineCert } =
+      this.deserializeSignatureFileContents(signatureFile);
+
+    // Verify type and authenticity of signing machine cert
+    const certDetails = await parseCert(signingMachineCert);
+    switch (artifact.type) {
+      case 'cvr_file': {
+        assert(
+          certDetails.component === 'central-scan' ||
+            certDetails.component === 'scan',
+          'Signing machine cert for CVR file should be a VxCentralScan or VxScan cert'
+        );
+        break;
+      }
+      case 'election_definition': {
+        assert(
+          certDetails.component === 'admin',
+          'Signing machine cert for election definition should be a VxAdmin cert'
+        );
+        break;
+      }
+      /* istanbul ignore next: Compile-time check for completeness */
+      default: {
+        throwIllegalValue(artifact.type);
+      }
+    }
+    await verifyFirstCertWasSignedBySecondCert(
+      signingMachineCert,
+      this.vxCertAuthorityCertPath
+    );
+
+    const signingMachinePublicKey = await extractPublicKeyFromCert(
+      signingMachineCert
+    );
+    await verifySignature({
+      message,
+      messageSignature,
+      publicKey: signingMachinePublicKey,
+    });
+  }
+
+  private serializeSignatureFileContents(
+    signatureFileContents: SignatureFileContents
+  ): Buffer {
+    const { signature, signingMachineCert } = signatureFileContents;
+    return Buffer.concat([
+      Buffer.from([signature.length]),
+      signature,
+      signingMachineCert,
+    ]);
```

Comment:
Settled on simple buffer concatenation instead of, say, a JSON file, because the latter made it too easy to accidentally add new unverified/unsigned fields.

I actually tried JSON out and myself made this mistake, accidentally moving the version number (which you'll see below) out of the signed message and into a top-level JSON field, only to remember that the new location would allow someone malicious to manually modify the version number and thereby influence parsing.